### ISSUE TITLE
Comprehensible error object for `connect_error` handler

### ISSUE THREAD

**Issue by andrejleitner** [socketio/socket.io-client#1523]
**Is your feature request related to a problem? Please describe.**
It is not possible to distinguish between errors in `connect_error` event listener (client side).
E.g. `Session ID unknown` will produce error with properties:
```js
{
   message: 'xhr poll error',
   type: 'TransportError',
   description: 400
}
```

**Describe the solution you'd like**
It would be useful to properly type the following error, make its properties comprehensible and add the reason too (e.g. Session ID unknown, Forbidden). 
```js 
interface SocketReservedEvents {
    connect: () => void;
    connect_error: (err: Error) => void;
    disconnect: (reason: Socket.DisconnectReason) => void;
}
```

**Comment by darrachequesne** [comment]
That's a great idea! Currently, we only use the `status` of the request: https://github.com/socketio/engine.io-client/blob/56af9c59b9f57946ac282182e0d9a53fcdad1d64/lib/transports/polling-xhr.ts#L204

But I think we could parse the value of the `responseText` attribute.

**Comment by andrejleitner** [comment]
Yep, I would move `status` code to separate property (that may be useful too) and fix the description with the real one.
